tests/test_unit/test_kinematics.py

```diff
@@ -182,3 +184,120 @@ def test_approximate_derivative_with_invalid_order(order):
     expected_exception = ValueError if isinstance(order, int) else TypeError
     with pytest.raises(expected_exception):
         kinematics._compute_approximate_time_derivative(data, order=order)
+
+
+class TestNavigation:
+    """Test suite for navigation-related functions in the kinematics module."""
+
+    @pytest.fixture
+    def mock_dataarray(self):
+        """Return a mock DataArray containing four known head orientations."""
+        time = [0, 1, 2, 3]
+        individuals = ["individual_0"]
+        keypoints = ["left_ear", "right_ear", "nose"]
+        space = ["x", "y"]
+
+        ds = xr.DataArray(
+            [
+                [[[1, 0], [-1, 0], [0, -1]]],  # time 0
+                [[[0, 1], [0, -1], [1, 0]]],  # time 1
+                [[[-1, 0], [1, 0], [0, 1]]],  # time 2
+                [[[0, -1], [0, 1], [-1, 0]]],  # time 3
+            ],
+            dims=["time", "individuals", "keypoints", "space"],
+            coords={
+                "time": time,
+                "individuals": individuals,
+                "keypoints": keypoints,
+                "space": space,
+            },
+        )
+        return ds
+
+    @pytest.fixture
+    def mock_dataarray_3D(self):
```

Comment:
Do we intend this function to be used in 3D data?

I am not sure it would make sense since we assume the animal is seen from either top-down or bottom-up view.

If you want to test that we catch when the input data array has incorrect number of spatial dimensions, I would make an `invalid_spatial_dimensions_data_array` fixture, that takes the `valid_data_array` fixture and adds an extra spatial dimension.

That way we reduce duplication and is easier to read